src/sentry/utils/pytest/selenium.py

```diff
@@ -335,6 +333,13 @@ def fin():
     return driver
 
 
+@pytest.fixture(scope='class')
+def browser(request, _browser):
+    if hasattr(request, 'cls'):
+        request.cls.browser = request.session.browser
+    request.session.browser.driver.delete_all_cookies()
```

Comment:
AFAIK this won't account for a number of other cleanup concerns, which will be forever expanding:

Two obvious ones to me:

- local storage
- web workers

Reply:
I've handled local storage. Web workers is something we'll have to make changes for if we decide to integrate them into sentry.

I think should we run into any issues like this we can make adjustments to the testing infrastructure.

Reply:
my concern is just that this can say the build is green even though things are broken because its using historical browser state/data in ways that the user might not understand

i think its worth asking if this is best for #engineering as a whole to shave off some time on one test suite when we can just distribute tests to run them in parallel

in my person experience its very risky to allow false build state, because the bigger the team is, the less likely an individual is going to understand the caveats of things

Reply:
Yea, we definitely want to avoid false positives. 

Once you start writing acceptance tests though, I would expect you start to understand how that infrastructure works, so if you write a failing test that you expect to work you would need to dig into.

What you're saying though is you could write a test that passes but only because some browser state is leaking across test cases?

I think we should be good on coverage clearing cookies and session/local storage.

Reply:
yeah thats what i was suggesting - my concern is you have to also keep up with browser features potentially which is tough since ideally the testing infrastructure doesnt have to be super aware

i would at least fully document the complexity of it as comments

Reply:
What I really wanted here would be to keep the chrome instance running, but create a new incognito window each time, and let chrome be responsible for knowing how to keep browser state ephemeral.

Reply:
fwiw incognito persists state between windows too within the same session

Reply:
Only if you don't close the incognito session.